toolbox/relnotes/main.go

```diff
@@ -0,0 +1,606 @@
+		// unbounded for hours
+		extraFlag = "--limit=30"
+	}
+
+	f, err := os.OpenFile(outputFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return err
+	}
+
+	defer f.Close()
+
+	f.WriteString(fmt.Sprintf("## State of %s branch\n", branch))
+
+	// Call script find_green_build to get CI job status
+	content, err := u.Shell(os.Getenv("GOPATH")+"/src/k8s.io/release/find_green_build", "-v", extraFlag, branch)
+	if err == nil {
+		f.WriteString(fmt.Sprintf("%sGOOD TO GO!%s\n\n", green, off))
+	} else {
+		f.WriteString(fmt.Sprintf("%sNOT READY%s\n\n", red, off))
+	}
+
+	f.WriteString("### Details\n```\n")
+	f.WriteString(content)
+	f.WriteString("```\n")
+
+	log.Printf("CI job status fetched.")
+	return nil
+}
+
+// createBody creates the general documentation, example and downloads table body for the
+// markdown file.
+func createBody(c *github.Client, f *os.File, releaseTag, branch, docURL, exampleURL, releaseTars string) {
+	var title string
+	if *preview {
+		title = "Branch "
+	}
+
+	if releaseTag == "HEAD" || releaseTag == branchHead {
+		title += branch
+	} else {
+		title += releaseTag
+	}
+
+	if *preview {
+		f.WriteString("**Release Note Preview - generated on " + time.Now().Format("Mon Jan  2 15:04:05 MST 2006") + "**\n")
+	}
+
+	f.WriteString("\n# " + title + "\n\n")
+	f.WriteString(fmt.Sprintf("[Documentation](%s) & [Examples](%s)\n\n", docURL, exampleURL))
+
+	if releaseTars != "" {
+		f.WriteString("## Downloads for " + title + "\n\n")
+		createDownloadsTable(f, releaseTag, "", releaseTars+"/kubernetes.tar.gz", releaseTars+"/kubernetes-src.tar.gz")
+		createDownloadsTable(f, releaseTag, "Client Binaries", releaseTars+"/kubernetes-client*.tar.gz")
+		createDownloadsTable(f, releaseTag, "Server Binaries", releaseTars+"/kubernetes-server*.tar.gz")
+		createDownloadsTable(f, releaseTag, "Node Binaries", releaseTars+"/kubernetes-node*.tar.gz")
+		f.WriteString("\n")
+	}
+}
+
+// createDownloadTable creates table of download link and sha256 hash for given file.
+func createDownloadsTable(f *os.File, releaseTag, heading string, filename ...string) {
+	var urlPrefix string
+
+	if *releaseBucket == "kubernetes-release" {
+		urlPrefix = "https://dl.k8s.io"
+	} else {
+		urlPrefix = "https://storage.googleapis.com/" + *releaseBucket + "/release"
+	}
+
+	if heading != "" {
+		f.WriteString(fmt.Sprintf("\n### %s\n", heading))
+	}
+
+	f.WriteString("\n")
+	f.WriteString("filename | sha256 hash\n")
+	f.WriteString("-------- | -----------\n")
+
+	files := make([]string, 0)
+	for _, name := range filename {
+		fs, _ := filepath.Glob(name)
+		for _, v := range fs {
+			files = append(files, v)
+		}
+	}
+
+	for _, file := range files {
+		fn := extractFileName(file)
+		sha, _ := u.GetSha256(file)
+		f.WriteString(fmt.Sprintf("[%s](%s/%s/%s) | `%s`\n", fn, urlPrefix, releaseTag, fn, sha))
+	}
+}
+
+// extractFileName takes a string and returns the file name after last '/'.
+func extractFileName(filePath string) string {
```

Comment:
You can use `filepath.Base()` https://golang.org/pkg/path/filepath/#Base

Reply:
Done.